### PR TITLE
Modernize Dependencies

### PR DESCRIPTION
**Pull request by versecafe**
This moves all dependencies to their newest stable release and updates any deprecations with two exceptions being `serialize-error` and `stack-trace` both of which have some CJS compatibility issues, `@msgpack/msgpack` is on the latest non beta release

```
@msgpack/msgpack 2.8.0   2.8.0  3.0.0-beta2 @logtail/edge dependencies https://msgpack.org/
@msgpack/msgpack 2.8.0   2.8.0  3.0.0-beta2 @logtail/node dependencies https://msgpack.org/
serialize-error  8.1.0   8.1.0  11.0.3      @logtail/core dependencies https://github.com/sindresorhus/serialize-error#readme
serialize-error  8.1.0   8.1.0  11.0.3      @logtail/edge dependencies https://github.com/sindresorhus/serialize-error#readme
serialize-error  8.1.0   8.1.0  11.0.3      @logtail/node dependencies https://github.com/sindresorhus/serialize-error#readme
stack-trace      0.0.10  0.0.10 1.0.0-pre2  @logtail/edge dependencies https://github.com/felixge/node-stack-trace
stack-trace      0.0.10  0.0.10 1.0.0-pre2  @logtail/node dependencies https://github.com/felixge/node-stack-trace
```

This also comes with Lerna + Nx which improves caching, features like lerna bootstrap are no longer needed they work directly within the pm making dependency tracking easier and creating a singular lock file, and plenty of other small changes to things such as formatting behaviors, performance, compatibility, etc.

## Breaking Changes

- No end user breaking changes
- CI for Node 16.x is no longer functional as Lerna v7 and on do not support anything below 18.x, but the end packages are 16.x compatible

### CODE REVIEW

**Comment by versecafe** [comment]
@PetrHeinz Can you give this a review when you have some time, thank you.

**Comment by PetrHeinz** [comment]
Thank you @versecafe for the PR and for your patience with the review. I really appreciate the work, description and commit separation. That should make the review easier for me 🙌

**Comment by versecafe** [comment]
@PetrHeinz Sorry for the ping, any update on the review?

**Comment by PetrHeinz** [comment]
Hi @versecafe, thanks for the ping 🙌 I got a few higher priority tasks ahead at the moment, so I can't really provide any ETA.

Thanks for understanding, your patience, and for the effort that went into the PR, I really appreciate it 🙏

**Comment by versecafe** [comment]
Thanks for the update, No rush just wanted to know if it was dropped as stale or only put on hold as lower priority.

File: packages/winston/src/winston.ts
Comment (versecafe) [inline packages/winston/src/winston.ts:1]: 👀

**Comment by PetrHeinz** [comment]
Hi @versecafe, thanks for your patience 🙏 

I've reviewed and tested the changes, I believe we're good to go for `v0.5.0` with this change 👍 Just made two tweaks to the changes.

Thank you very much for taking the time and effort to improve the library! 🙌

**Comment by PetrHeinz** [comment]
`v0.5.0` released and re-tested, everything works just nicely 🎉 

Thanks a lot, @versecafe!

**Pull request merged**